docs/guides/distribution/updater.md

```diff
@@ -2,220 +2,128 @@
+      ],
+      "dialog": true,
+      "pubkey": "YOUR_UPDATER_SIGNATURE_PUBKEY_HERE"
+    }
   }
-} catch (error) {
-  console.log(error)
 }
 ```
 
-### Events
-
-:::caution
-
-You need to _disable the built-in dialog_ in your [tauri configuration](#configuration); Otherwise, events aren't emitted.
-
-:::
-
-To know when an update is ready to be installed, you can subscribe to these events:
-
-#### Initialize updater and check if a new version is available
-
-##### If a new version is available, the event `tauri://update-available` is emitted.
-
-Event: `tauri://update`
-
-#### Rust
-
-```rust
-window.emit("tauri://update".to_string(), None);
-```
-
-#### Javascript
-
-```js
-import { emit } from '@tauri-apps/api/event'
-emit('tauri://update')
-```
-
-#### Listen to the New Update Available event
-
-Event: `tauri://update-available`
-
-Emitted data:
-
-```
-version    Version announced by the server
-date       Date announced by the server
-body       Note announced by the server
-```
-
-#### Rust
+The required keys are `"active"`, `"endpoints"` and `"pubkey"` to enable the updater. `"dialog"` is optional and will default to `true` if not set.
 
-```rust
-window.listen("tauri://update-available".to_string(), move |msg| {
-  println!("New version available: {:?}", msg);
-})
-```
+`"active"` must be a boolean. By default it's set to false.
 
-#### Javascript
+`"endpoints"` must be an array of updater endpoint URLs as strings. TLS is enforced in production mode.<br/>
+Each updater URL can contain the following variables allowing you to determine [server-side](#update-server-json-format) if an update is available:
 
-```js
-import { listen } from '@tauri-apps/api/event'
-listen('tauri://update-available', function (res) {
-  console.log('New version available: ', res)
-})
-```
+- `{{current_version}}`: The version of the app that is requesting the update.
+- `{{target}}`: The operating system name (one of `linux`, `windows` or `darwin`).
+- `{{arch}}`: The architecture of the machine (one of `x86_64`, `i686`, `aarch64` or `armv7`).
 
-#### Emit Install and Download events
+`"pubkey"` must be a valid public key generated with Tauri's CLI [above](#signing-updates).
 
-You need to emit this event to initialize the download and listen to the [install progress](#listen-install-progress).
+`"dialog"` if present must ba a boolean. By default it's set to true. If enabled, updater [events](#events) will be disabled as the built-in dialog handles everything. If you need custom events, you must turn off the built-in dialog.
 
-Event: `tauri://update-install`
+## Update artifacts
 
-#### Rust
+Tauri's bundler will automatically generate and sign update artifacts once the updater is correctly configured and enabled.
 
-```rust
-window.emit("tauri://update-install".to_string(), None);
-```
+Before building your app, you need to set environment variables for the private key and password:
```

Comment:
I'd add a code snippet here of how you set the environment variables

Reply:
hmm, the problem with that is the amount of different ways to do that and platform inconsistencies 🤔

Reply:
i added instructions to set the env var for the current session, is that ok? (just pushed it so may need a while to build)